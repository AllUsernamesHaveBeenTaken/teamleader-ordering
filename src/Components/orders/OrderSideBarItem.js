import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import { elevation, transition } from '../../Utilities';

const OrderSideBarItem = ({ selectOrder, id }) => (
  <MenuItemWrapper data-testid="menu-item" onClick={() => selectOrder(id)}>
    <MenuItemTitle data-testid="menu-item-title">{`Order No. ${id}`}</MenuItemTitle>
  </MenuItemWrapper>
);

export default OrderSideBarItem;

OrderSideBarItem.propTypes = {
  id: PropTypes.string.isRequired,
  selectOrder: PropTypes.func.isRequired,
};

const MenuItemWrapper = styled.div`
  overflow: auto;
  border-bottom: 1px solid black;
  cursor: pointer;
  ${transition({})};
  &:hover {
    ${elevation[4]};
  }
`;

const MenuItemTitle = styled.h1`
  margin: 0;
  padding: 1rem;
`;
