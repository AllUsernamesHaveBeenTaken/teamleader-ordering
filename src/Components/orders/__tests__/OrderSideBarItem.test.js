import React from 'react';
import { render, cleanup, fireEvent } from 'react-testing-library';
import OrderSideBarItem from '../OrderSideBarItem';

afterEach(() => {
  cleanup();
  console.error.mockClear();
});

console.error = jest.fn();

const orderId = '1';

const selectOrder = jest.fn();

// OrderSideBarItem's props = id, selectOrder
test('<OrderSideBarItem />', () => {
  const { getByTestId } = render(
    <OrderSideBarItem id={orderId} selectOrder={selectOrder} />,
  );

  const menuItem = getByTestId('menu-item');

  expect(menuItem).toBeTruthy();

  expect(getByTestId('menu-item').textContent).toBe(`Order No. ${orderId}`);

  fireEvent.click(menuItem);
  expect(selectOrder).toHaveBeenCalledTimes(1);
  expect(selectOrder).toBeCalledWith(orderId);
});
