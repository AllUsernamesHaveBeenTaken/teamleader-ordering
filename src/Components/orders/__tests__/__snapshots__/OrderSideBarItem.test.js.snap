// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OrderSideBarItem /> 1`] = `
<div
  class="sc-bdVaJa hLZleQ"
  data-testid="menu-item"
>
  <h1
    class="sc-bwzfXH hDweOk"
    data-testid="menu-item-title"
  >
    Order No. 1
  </h1>
</div>
`;
